covid.py

```diff
@@ -1 +1,112 @@
-#this is comment
+import argparse
+import ast
+import json
+import requests
+import os
+
+cache_file = "data.txt"
+
+
+def get_parsed_arguments():
+    my_parser = argparse.ArgumentParser(
+        description="By giving name of country you will get "
+                    "percentage of people recovered and still sick from all population"
+    )
+    my_parser.add_argument(
+        "--country",
+        action="store",
+        required=True,
+        help="Name of country",
+    )
+    return my_parser.parse_args()
+
+
+def compute_answer(population, confirmed_cases, recovered_people, dead_people):
+    percent_of_recovered = round(100 * recovered_people / population, 1)
+
+    percent_of_sick = round(
+        100 * (confirmed_cases - recovered_people - dead_people) / population, 1
+    )
+
+    return json.dumps(
+        {"recovered": percent_of_recovered, "sick": percent_of_sick}, indent=1
+    )
+
+
+def call_covid_api(country_name):
+    response = requests.get(
+        url="https://covid-api.mmediagroup.fr/v1/cases?country=" + country_name
+    )
+    return response
+
+
+def check_is_cache_empty():
+    if os.stat(cache_file).st_size == 0:
+        return True
+    else:
+        return False
+
+
+def save_response_to_cache(response):
+    f = open(cache_file, "w")
+    f.write(str(response.json()))
+    f.close()
+
+
+def get_caches_content():
+    f = open(cache_file, "r")
+    return ast.literal_eval(f.read())
+
+
+def check_if_calling_api_is_needed(country):
+    if not check_is_cache_empty():
+        cached_info = get_caches_content()
+        try:
+            if cached_info.get('All').get('country') == country:
+                return False
+            else:
+                return True
+        except AttributeError:
+            return True
+    else:
+        return True
+
+
+def main():
+    args = get_parsed_arguments()
+
+    if check_if_calling_api_is_needed(args.country.capitalize()):
+        covid_api_response = call_covid_api(args.country.capitalize())
+        save_response_to_cache(covid_api_response)
+
+        if covid_api_response.ok:
+
+            all_data = covid_api_response.json()
+
+        else:
+            print(
+                "Oh no, unfortunately, some external error occurred. Please try again later"
+            )
+
+    else:
+        all_data = get_caches_content()
+
+    country_data = all_data.get("All")
```

Comment:
please move all the stuff to a separate function: `get_covid_data`. At top level function main, most likely we won't care if it's API or cached. This is low-level information

Reply:
I rogered that. I'm getting back to work to correct the code.